src/app/components/DateDisplay.tsx

```diff
@@ -16,22 +16,21 @@ export default class DateDisplay extends React.PureComponent<DateDisplayProps> {
     render() {
         const { date, showMonth=false, showDay=false, showYear=false, showTime=false } = this.props;
         
-        let dateFormat = "";
-        if (showMonth && showDay && showYear && showTime) {
-            dateFormat = "MMM D, YYYY H:mm";
+        let dateFormats = [];
+        if (showMonth) {
+            dateFormats.push("MMM");
         }
-        else if (showMonth && showDay && showYear && !showTime) {
-            dateFormat = "MMM D, YYYY";
+        if (showDay) {
+            dateFormats.push("D");
```

Comment:
no comma in the day format?

Reply:
I think the comma would have to be conditional (i.e. only show the comma if you are also showing a month and something after the comma) so for simplicity, we could probably leave it out for now